### DISCUSSION TITLE
mushr slam mapping

### DISCUSSION THREAD

**Discussion by park-dongjun** [prl-mushr/mushr#76] [Q&A]
![Screenshot from 2022-04-01 13-21-44](https://user-images.githubusercontent.com/102731549/161195097-4ef8f880-defd-4f3d-8ef1-8fb6ee488e63.png)

![Screenshot from 2022-04-01 13-15-37](https://user-images.githubusercontent.com/102731549/161195100-3e577c48-410e-4ef6-b8ec-da40ba0a34c9.png)
![Screenshot from 2022-04-01 13-16-32](https://user-images.githubusercontent.com/102731549/161195104-8e0e8e13-1e53-4bf8-a19e-c33d7dafcd7c.png)
![Screenshot from 2022-04-01 13-17-00](https://user-images.githubusercontent.com/102731549/161195105-d7e99d50-24cc-45c4-aadd-681de00783b2.png)

Following the tutorials, I finished all the installation process and ram slam, but as shown in the picture above, RobotModel error occurs and mapping does not work. Plz help me

**Comment by palanc** [comment]
Hi @park-dongjun - Let's see if getting the model to show up helps first. I am guessing that the model is not showing up because rviz does not know where to find the mesh models. In the terminal that launches rviz on your desktop/laptop, have you source'd the workspace that contains the MuSHR stack?

**Reply by park-dongjun** [reply]
![Screenshot from 2022-04-07 18-53-33](https://user-images.githubusercontent.com/102731549/162173706-b4c780b2-2ccd-43e9-84a3-44b0b7c90c62.png)
![Screenshot from 2022-04-07 18-54-11](https://user-images.githubusercontent.com/102731549/162173728-98a38a0e-a0ea-486b-892f-e587e1372aaf.png)
![Screenshot from 2022-04-07 18-54-16](https://user-images.githubusercontent.com/102731549/162173735-093c23f7-d9f6-4344-a1bd-2233a81132bd.png)

Teleop works just fine. You can see capture above. 
But when I run gmapping, I get the error.
We suspect that /car/map and /map conflict with each other, causing this problem.

These are our problem at rviz.
![Screenshot from 2022-04-07 18-47-45](https://user-images.githubusercontent.com/102731549/162174011-2b8f84dd-a2c2-431f-a45d-887947d37b86.png)

![Screenshot from 2022-04-07 18-48-02](https://user-images.githubusercontent.com/102731549/162174029-0c5ba875-99fe-47f3-b537-7a9c868cdc99.png)

When we just launch teleop rviz can get model, but when we launch gmapping there are big problem.

**Reply by park-dongjun** [reply]
I think there are two publisher so they conflict each other please check and if you want to get more information , tell us

**Reply by palanc** [reply]
In rviz, Displays -> Global Options -> Fixed Frame should be set ```/car/map```. Also, Displays -> Map -> Topic should also be set to ```/car/map```. Have you tried that?

**Comment by park-dongjun** [comment]
![1](https://user-images.githubusercontent.com/102731549/162344223-d15281d5-c2ca-461c-a1a2-1cdd56dc1d8a.png)

Yes. I changed Global Options -> Fixed Frame to /car/map, Displays -> Map -> Topic to /car/map as pictured.

Even after doing so, the same problem occurs. What should I do?..

**Reply by palanc** [reply]
I think it's okay if the robot model doesn't show up. With these latest settings, does the map progressively get built in rviz if you slowly teleoperate the car around?

**Reply by park-dongjun** [reply]
Laser scan works, but gmapping does not work Even if I set that setting in several places, the result is always the same as in the picture above. If you set it to /car/map at first, it appears as if it is a mapping, but it seems that the mapping does not work as the map does not spread after that.

**Reply by palanc** [reply]
That is strange that it works initially but then seems to stop working. How long are you waiting to see if the map updates (it could take on the order of ~10 seconds)? When you teleoperate the car around, do the lidar scans shown in rviz change accordingly? Are there any errors in the terminal that you launched slam.launch from or any of the other terminals?

**Reply by park-dongjun** [reply]
I waited about 10 minutes to see the map updated, but it didn't update.
Lidar does scan. If I move it with a joystick, car moves, and rviz recognize the car move. but it can't mapping.
There is no error in slam.launch

**Reply by palanc** [reply]
@mcdoerr @MarkusSchiffer Any ideas of what might be wrong?

**Reply by mcdoerr** [reply]
Hi, have you ensured that the ROS_IP on your computer is set correctly and the ROS_MASTER_URI is set to `http://ROBOT_IP:11311
 `? I don't see these commands in your terminals before you run rviz. Might be not be connecting right.

**Reply by mcdoerr** [reply]
I'm not sure why the map won't continue to update. That's strange behavior. When slam is running, can you run `rostopic list`?

**Reply by park-dongjun** [reply]
![bashrc](https://user-images.githubusercontent.com/102731549/163341175-c25f4dac-a844-4ab7-b771-ed841112a71e.png)
![laptop_IP](https://user-images.githubusercontent.com/102731549/163341183-389d6750-f8ae-4239-90a9-b0d6d713ac5e.png)
![MuSHR_IP](https://user-images.githubusercontent.com/102731549/163341188-7b312bf1-474a-48a1-8086-319b4a03c272.png)
![rostopiclist](https://user-images.githubusercontent.com/102731549/163341190-83f3588d-b8d3-4bf8-abbb-5d80f0fefd2d.png)

Here is the ROS Topic list, laptop IP, and MuSHR IP.

**Reply by mcdoerr** [reply]
That all looks as expected. I bet that you're right about the `/car/map` and `/map` topic conflicting with eachother. Let's see if we can turn off the `/map` topic. Can you run `rostopic info /map` when all the nodes are running so we can see what is publishing that? There might be multiple publishers. 

Also, can you make sure your code is up to date? We removed some code that was causing multiple map servers to be running, just want to make sure that isn't running.

**Reply by mcdoerr** [reply]
Also, FYI, I have run into the issue with the robot model not showing up as you pictured. I think it's a transform issue and slam worked even when the robot was white, so I don't think they are related.

**Reply by park-dongjun** [reply]
We also think that /car/map and /map collide. A photo of rostopic info /map is attached below. please help me how to solve this problem

**Comment by park-dongjun** [comment]
![Screenshot from 2022-04-14 17-36-46](https://user-images.githubusercontent.com/102731549/165695622-3e0dba11-a4a2-4b7e-84c6-78c5f8fa6fea.png)

This is rostopic info /map and carmap

**Reply by Choiki1228** [reply]
Did you solve this, I'm having the same problem as you. Can you help me solve this problem or give me some advice.
